pkg/controller/compute/workload/workload.go

```diff
@@ -271,6 +271,15 @@ func (r *Reconciler) _delete(instance *computev1alpha1.Workload, client kubernet
 		return r.fail(instance, errorDeleting, err.Error())
 	}
 
+	// delete resources secrets
+	for _, resource := range instance.Spec.Resources {
+		secretName := util.IfEmptyString(resource.SecretName, resource.Name)
+		err = client.CoreV1().Secrets(ns).Delete(secretName, &metav1.DeleteOptions{})
+		if err != nil {
```

Comment:
Consider checking for "non-existent", otherwise already deleted secret. 
```suggestion
		if err := client.CoreV1().Secrets(ns).Delete(secretName, &metav1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
```